janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java

```diff
@@ -1482,6 +1482,7 @@ private void releaseTransaction() {
         isOpen = false;
         graph.closeTransaction(this);
         vertexCache.close();
+        indexCache.close();
```

Comment:
Next data structures may explicitly cleared too 

```
        if (deletedRelations != EMPTY_DELETED_RELATIONS) {
            deletedRelations.clear();
        }
        if (uniqueLocks != UNINITIALIZED_LOCKS){
            uniqueLocks.clear();
        }
        addedRelations.clear();
```

Reply:
@li-boxuan @mad 
See my comment here: https://github.com/JanusGraph/janusgraph/pull/2413/files#r571375536

Reply:
Can't we just change that to?:
addedRelations = null;
deletedRelations = null;
uniqueLocks = null;

Also, I didn't check why we need to clear those data structures, it's just quick observations. My point is that some implementations of 'clear' method are really inefficient. Again, didn't check the complexity of the above clear methods